x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/Layout.java

```diff
@@ -108,16 +108,27 @@ public Layout build() {
             int numberOfChannels = 0;
             for (ChannelSet set : channels) {
                 int channel = numberOfChannels++;
-                for (NameId id : set.nameIds) {
-                    ChannelAndType next = new ChannelAndType(channel, set.type);
-                    ChannelAndType prev = layout.put(id, next);
-                    // Do allow multiple name to point to the same channel - see https://github.com/elastic/elasticsearch/pull/100238
-                    // if (prev != null) {
-                    // throw new IllegalArgumentException("Name [" + id + "] is on two channels [" + prev + "] and [" + next + "]");
-                    // }
+                if (set != null) {
```

Comment:
Please add some comments on why the set could be null - feels like a bug that caused the initialization to be skipped.

Reply:
I thought it was a bug as well, but looking at the implementation I'm not completely sure, it could be on purpose

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/DefaultLayout.java#L42

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/DefaultLayout.java#L65

maybe @nik9000 has a clue

Anyway, it happens because multiple projections point to the same agg (so to the same channel) and the builder inverse has some empty positions. I'll add a comment